src/controllers/Social.ts

```diff
@@ -38,6 +44,38 @@ export default class Social {
    * private methods
    */
 
+  /**
+   * Gets the access key view.
+   * @param {Account} account - an initialized account.
+   * @param {string | utils.PublicKey} publicKey - the public key of the access key to query.
+   * @returns {Promise<AccessKeyView | null>} a promise that resolves to the access key view or null if the access key
+   * for the given public key does not exist.
+   * @private
+   */
+  private async _accessKeyView(
```

Comment:
Querying the blockchain for access key sometime also cause unnecessary delays if you are just sequentially calling the chain from the same system (UI or a script), near-js/accounts handles this by keeping a local cache of the access key. 

Can we try something like this. 
See this here - https://github.com/near/near-api-js/blob/master/packages/accounts/src/account.ts#L237 

We can copy the whole function as well. 

That being said, I am okay to merge this right now and handle the access key cache in a separate PR.